pkg/ingress/controller/controller.go

```diff
@@ -414,36 +414,35 @@ func (c *Controller) processItem(event Event) error {
 	ing := event.Obj.(*extv1beta1.Ingress)
 	key := fmt.Sprintf("%s/%s", ing.Namespace, ing.Name)
 
+	var verb string
+
 	switch event.Type {
 	case CreateEvent:
-		log.WithFields(log.Fields{"ingress": key}).Info("ingress created, will create openstack resources")
-
-		if err := c.ensureIngress(ing); err != nil {
-			utilruntime.HandleError(fmt.Errorf("failed to create openstack resources for ingress %s: %v", key, err))
-			c.recorder.Event(ing, apiv1.EventTypeWarning, "Failed", fmt.Sprintf("Failed to create openstack resources for ingress %s: %v", key, err))
-		} else {
-			c.recorder.Event(ing, apiv1.EventTypeNormal, "Created", fmt.Sprintf("Ingress %s", key))
-		}
+		verb = "create"
 	case UpdateEvent:
-		log.WithFields(log.Fields{"ingress": key}).Info("ingress updated, will update openstack resources")
-
-		if err := c.ensureIngress(ing); err != nil {
-			utilruntime.HandleError(fmt.Errorf("failed to update openstack resources for ingress %s: %v", key, err))
-			c.recorder.Event(ing, apiv1.EventTypeWarning, "Failed", fmt.Sprintf("Failed to update openstack resources for ingress %s: %v", key, err))
-		} else {
-			c.recorder.Event(ing, apiv1.EventTypeNormal, "Updated", fmt.Sprintf("Ingress %s", key))
-		}
+		verb = "update"
 	case DeleteEvent:
-		log.WithFields(log.Fields{"ingress": key}).Info("ingress has been deleted, will delete openstack resources")
+		log.WithFields(log.Fields{"ingress": key}).Info("ingress has been deleted, deleting openstack resources")
 
 		if err := c.deleteIngress(ing.Namespace, ing.Name); err != nil {
 			utilruntime.HandleError(fmt.Errorf("failed to delete openstack resources for ingress %s: %v", key, err))
 			c.recorder.Event(ing, apiv1.EventTypeWarning, "Failed", fmt.Sprintf("Failed to delete openstack resources for ingress %s: %v", key, err))
-		} else {
-			c.recorder.Event(ing, apiv1.EventTypeNormal, "Deleted", fmt.Sprintf("Ingress %s", key))
+			return err
 		}
+
+		c.recorder.Event(ing, apiv1.EventTypeNormal, "Deleted", fmt.Sprintf("Ingress %s", key))
+		return nil
+	}
+
+	log.WithFields(log.Fields{"ingress": key}).Infof("ingress %sd, will update openstack resources", verb)
```

Comment:
I personally don't think it's a good idea for the refactoring in this patch, although there is some code duplication in the original implementation, it's more clear for the logic.